k9mail/src/main/java/security/planck/ui/removeaccount/RemoveAccountViewModel.kt

```diff
@@ -0,0 +1,94 @@
+package security.planck.ui.removeaccount
+
+import android.app.Application
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.fsck.k9.Account
+import com.fsck.k9.K9
+import com.fsck.k9.Preferences
+import com.fsck.k9.controller.MessagingController
+import com.fsck.k9.planck.DispatcherProvider
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import timber.log.Timber
+import javax.inject.Inject
+
+private const val MINIMUM_DELAY_FOR_ACCOUNT_DELETION = 500L
+
+@HiltViewModel
+class RemoveAccountViewModel @Inject constructor(
+    private val context: Application,
+    private val preferences: Preferences,
+    private val controller: MessagingController,
+    private val dispatcherProvider: DispatcherProvider,
+) : ViewModel() {
+    private lateinit var account: Account
+
+    private val stateLiveData: MutableLiveData<RemoveAccountState> =
+        MutableLiveData(RemoveAccountState.Idle)
+    val state: LiveData<RemoveAccountState> = stateLiveData
+
+    fun initialize(uuid: String) {
+        viewModelScope.launch {
+            account = preferences.getAccount(uuid) ?: let {
+                stateLiveData.value = RemoveAccountState.AccountNotAvailable(uuid)
+                return@launch
+            }
+            stateLiveData.value =
+                RemoveAccountState.RemoveAccountConfirmation(account.description ?: account.email)
+        }
+    }
+
+    fun positiveAction() {
+        when (stateLiveData.value) {
+            is RemoveAccountState.RemoveAccountConfirmation ->
+                viewModelScope.launch {
+                    removeAccount()
+                }
+
+            is RemoveAccountState.Done ->
+                stateLiveData.value = RemoveAccountState.Finish(true)
+
+            is RemoveAccountState.AccountNotAvailable ->
+                stateLiveData.value = RemoveAccountState.Finish(true)
+
+            else -> error("unexpected state: ${stateLiveData.value}")
```

Comment:
I think we can write just Account not available here

Reply:
'when' is actually like Java's 'switch', and 'else' is like Java's 'default'. 'error' throws an IllegalStateException. So this is more of an error for the developer, we only expect to have this method called in these states.

Reply:
where is this string displayed?
> unexpected state: ${stateLiveData.value}

does it end up in any log file? Or is it displayed in any UI?